File: src/Chicken.cpp
#include "Chicken.hpp"
#include <iostream>

using namespace std;

Chicken::Chicken() :FarmAnimal(false, true, false, 10,0, 0) {
}

Chicken::Chicken(int x, int y):FarmAnimal(false, true, false, 10,x, y) {
}

void Chicken::Bersuara() //Menghasilkan suara Chicken
{
    cout<<"Petok Petok"<<endl; 
}

char Chicken::render() //Menghasilkan representasi dari Chicken
{
    return 'A';
}


FarmProduct* Chicken::interact() //Berinteraksi dengan Chicken menghasilkan ChickenEgg
{
    return new ChickenEgg();
} 

File: src Java/Chicken.java
public class Chicken extends FarmAnimal {
    /**
     * Default constructor
     */
    public Chicken(){
        super(false, true, false, 10, 0, 0);
    }

    /**
     * User-defined constructor
     * @param x posisi X hewan yang diinginkan
     * @param y posisi Y hewan yang diinginkan
     */
    public Chicken(int x, int y){
        super(false, true, false, 10, x, y);
    }

    /**
     * Mencetak representasi suara ayam
     */
    public void bersuara(){
        System.out.println("Petok petok");
    }

    /**
     * Mengembalikan karakter representasi ayam
     * @return 'A'
     */
    public char render(){
        return 'A';
    }

    /**
     * memberikan FarmProduct ketika pemain berinteraksi dengan ayam
     * @return objek ChickenEgg
     */
    public FarmProduct interact(){
        return new ChickenEgg();
    }
}